fem/dfem/tuple.hpp

```diff
@@ -0,0 +1,853 @@
+// Copyright (c) 2010-2025, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+#pragma once
+
+// This is serac's tuple implementation
+
+#include "../../general/backends.hpp"
+#include <utility>
+
+namespace mfem::experimental
+{
+
+/**
+ * @tparam T the types stored in the tuple
+ * @brief This is a class that mimics most of std::tuple's interface,
+ * except that it is usable in CUDA kernels and admits some arithmetic operator overloads.
+ *
+ * see https://en.cppreference.com/w/cpp/utility/tuple for more information about std::tuple
+ */
+template <typename... T>
+struct tuple
+{
+};
+
+/**
+ * @brief Type that mimics std::tuple
+ *
+ * @tparam T0 The first type stored in the tuple
+ */
+template <typename T0>
+struct tuple<T0>
+{
+   T0 v0;  ///< The first member of the tuple
+};
```

Comment:
I don't actually know, but can you do this recursively to obtain arbitrary length? Something along the lines of this pseudo-code...
```c++
template<typename... T>
struct tuple {
};

template<>
struct tuple{} { };

template<T0, typename... T>
struct tuple<T0,T...>  {
    T0 head;
     struct tuple<T...> tail;
};

template<int i, typename... T>
constexpr auto get()
{
    static_assert(i >= 0, "Bad get");
   if constexpr (i == 0) {
      return head; 
    }
    else {
      return get<i-1>(tail);
    } 
}
```

Reply:
There are problems with the initializers. :-(

Reply:
I didn't test this enough. It has trouble with curly brace initialization.
```c++
tuple<foo,bar,baz> example { 1, 342, 4};
```